ow_faults/src/FaultInjector.cpp

```diff
@@ -103,14 +109,59 @@ void FaultInjector::antennaeTiltFaultCb(const std_msgs::Float64& msg){
                         m_faultTiltValue, m_fault_ant_tilt_pub );
 }
 
-void FaultInjector::setPowerTemperatureFaultValue(bool getTempBool){
-  float thermal_val;
-  if (isnan(powerTemperatureOverloadValue)) {
-    thermal_val =  50.0 + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(90.0-50.0)));
-    powerTemperatureOverloadValue = thermal_val;
-  } else if (!getTempBool) {
-    powerTemperatureOverloadValue = NAN;
+float FaultInjector::getRandomFloatFromRange( float min_val, float max_val){
+  return min_val + (max_val - min_val) * (rand() / float(RAND_MAX));
+}
+
+void FaultInjector::publishPowerSystemFault(bool fault){
+  //power
+  ow_faults::PowerFaults power_faults_msg;
+  ComponentFaults hardwareFault =  ComponentFaults::Hardware;
+  //system
+  std::bitset<10> systemFaultsBitmask{};
+  ow_faults::SystemFaults system_faults_msg;
+
+  systemFaultsBitmask |= systemFaultsBitset;
+  //update if fault
+  if (fault) {
+    //system
+    systemFaultsBitmask |= isPowerSystemFault;
+    systemFaultsBitset = systemFaultsBitmask;
+    //power
+    setComponentFaultsMessage(power_faults_msg, hardwareFault);
   }
+  //publish
+  setBitsetFaultsMessage(system_faults_msg, systemFaultsBitmask);
+  m_fault_status_pub.publish(system_faults_msg);
+  m_power_fault_status_pub.publish(power_faults_msg);
+
+}
+
+void FaultInjector::powerTempListener(const std_msgs::Float64& msg)
+{
+  bool fault = false;
+  if(m_faults.thermal_power_failure && msg.data > 50){
```

Comment:
Magic number 50 - don't know what it means, a named constant would clarify.

Reply:
good point, will update